src/RankingAdapter.ts

```diff
@@ -77,33 +79,43 @@ export class RankingAdapter {
 
       // console.log('reuse the data array')
     } else {
-      // console.log('update the data array');
       // refresh the data array
       this.data = null;
       this.oldAttributes = this.getDisplayedAttributes();
 
-      const databaseData = new Array();
+      const databaseData = [];
 
       const scoreCols = this.getScoreColumns();
       const scoresData = [].concat(...scoreCols.map((col) => this.getScoreData(col.desc)));
 
-      this.oldOrder = this.getItemOrder();
+      this.oldOrder = this.getItemOrder(); // [3, 5, 6, 7] -> [[0,3], [1,5], ...]
+      const orderMap = new Map<number, number>(); // index, old-order
+      this.oldOrder.forEach((order, i) => orderMap.set(order, i));
+
+      const groups = this.getRanking().getGroups();
+      const groupIndexArray = groups.map((g) => {
+        const groupMap = new Map<number, number>();
+        g.order.forEach((order, i) => {groupMap.set(order, i);});
+        return groupMap;
+      });
+
       this.oldSelection = this.getSelectionUnsorted();
 
       this.provider.data.forEach((item, i) => {
-        const index = this.oldOrder.indexOf(i);
+        const index = orderMap.get(i);
         item[RankingAdapter.RANK_COLUMN_ID] = index >= 0 ? index : Number.NaN; //NaN if not found
 
+
         // include wether the row is selected
-        item[RankingAdapter.SELECTION_COLUMN_ID] = this.oldSelection.includes(i) ? 'Selected' : 'Unselected'; // TODO compare perfomance with assiging all Unselected and then only set those from the selection array
-        const groupIndex = this.getRanking().getGroups().findIndex((grp) => grp.order.indexOf(i) >= 0);
-        const groupName = groupIndex === -1 ? 'Unknown' : this.getRanking().getGroups()[groupIndex].name;
+        item[RankingAdapter.SELECTION_COLUMN_ID] = this.oldSelection.includes(i) ? 'Selected' : 'Unselected';
+        const groupIndex = groupIndexArray.findIndex((map) => map.has(i));
```

Comment:
If I understand correctly, you don't need to store a lot of maps for each index in `groupIndexArray`, since you only want to know if `i` is included in one of the groups.

Probably it is sufficient to write in line 95ff the following:

```ts
      const groupIndexArray = groups.map((g) => {
        return g.order.map((order) => order);
      });
```

And then use it here with [Array.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes):

```suggestion
        const groupIndex = groupIndexArray.findIndex((groupIndex) => groupIndex.includes(i));
```

Reply:
Thanks for the suggestion! Using a map is redundant.